btcec/schnorr/musig2/context.go

```diff
@@ -548,6 +573,129 @@ func (s *Session) RegisterPubNonce(nonce [PubNonceSize]byte) (bool, error) {
 	return haveAllNonces, nil
 }
 
+// GenerateAdaptorSecret generates an adaptor secret key. This must be called
+// after the public nonces have been registered for all signers, because
+// the validity of the adaptor secret key depends on the combined nonce. It
+// must also be called before any partial signatures are generated or provided,
+// because the validation of the signatures depends on the adaptor point.
+func (s *Session) GenerateAdaptor(msg [32]byte) (*btcec.ModNScalar, *btcec.JacobianPoint, error) {
```

Comment:
nit: doc was added after this PR was opened, but we try to wrap before 80 characters to help with review and readability in narrow/multi-column environments: https://github.com/btcsuite/btcd/blob/master/docs/code_formatting_rules.md#80-character-line-length